src/pyduktape.c

```diff
@@ -44,6 +44,8 @@ static PyObject *DukPy_eval_string(PyObject *self, PyObject *args) {
     }
 
     ctx = get_context_from_capsule(pyctx);
+    Py_XDECREF(pyctx);
```

Comment:
I think we should decrement the reference at the end of the function (on exit points).
The `pyctx` will call `duk_destroy_heap` through `context_destroy` when the references are gone, and thus will destroy the `ctx` which are are using below.

As far as we use `ctx` then we should not decrement `pyctx`. So I think `Py_XDECREF(pyctx)` should only go before a `return` within this method.

Reply:
Would you like me to post a new pull request tomorrow for this?

Reply:
Yes please, it would be great if you can update the PR according to the suggestion 👍